### PR TITLE
Potential fix for #1683

### PR DESCRIPTION
**Pull request by ghost**
The marker colshape is no longer attached to the marker, instead the hit point is updated. At least, this is how it's done in pickup code. It seems to work for pickups, so it shouldn't cause any problem for markers either. I tested it, and the marker colshape hit events are triggered properly.

EDIT: yes, the commit history is messed up. Fixing it.

### CODE REVIEW

**Comment by Dutchman101** [comment]
Nice.. i hope this will be it.

But do you have any idea why a crash like this would be unhandled (no dump, the client just goes poof)? It might be worth looking into that on the sidelines.

**Comment by ghost** [comment]
> But do you have any idea why a crash like this would be unhandled (no dump, the client just goes poof)? It might be worth looking into that on the sidelines.

This is an old issue. I have a theory: If an exception occurs in the exception handler code, then the dump won't be generated. If could somehow reproduce a crash where there's no dump file, then I might be able to track the bug.

File: Client/mods/deathmatch/logic/CClientMarker.cpp [hunk 1]
```diff
@@ -48,6 +50,7 @@ CClientMarker::~CClientMarker()
     // Remove the colshape
     if (m_pCollision)
         delete m_pCollision;
+    m_pCollision = nullptr;
```
Comment (qaisjp) [inline Client/mods/deathmatch/logic/CClientMarker.cpp]:
```suggestion
    SAFE_DELETE(m_pCollision);
```

**Review by qaisjp — APPROVED** [review]

**Comment by Pirulax** [comment]
> > But do you have any idea why a crash like this would be unhandled (no dump, the client just goes poof)? It might be worth looking into that on the sidelines.
> 
> This is an old issue. I have a theory: If an exception occurs in the exception handler code, then the dump won't be generated. If could somehow reproduce a crash where there's no dump file, then I might be able to track the bug.

We use google breakpad for crash exception handling, dont we? Maybe we should just update it.

Edit: I've looked it up on google("google breakpad no crashdump") and found [this](https://groups.google.com/a/chromium.org/g/chromium-discuss/c/3J69EcSz1Xo?pli=1). They basically say, if its an OOM(out of mem) issue the crash handler itself may not have enough address space to mmap the things it needs. Knowing how much memory MTA uses on a regular basis(2-3gb), i see this as a possibility. 

Edit2: People there suggest to use `crashpad` instead.

**Pull request merged**